File: src/main/java/com/wooteco/wiki/domain/Log.java
package com.wooteco.wiki.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor
@Getter
@NoArgsConstructor
@Builder
public class Log {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long logId;
    private String title;
    private String contents;
    private String writer;
    private Long documentBytes;
    private LocalDateTime generateTime;
    @ManyToOne
    private Member member;
}

File: src/main/java/com/wooteco/wiki/domain/Log.kt
package com.wooteco.wiki.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.time.LocalDateTime

@Entity
class Log(
    var title: String = "",
    var contents: String = "",
    var writer: String = "",
    var documentBytes: Long = 0,
    var generateTime: LocalDateTime = LocalDateTime.now(),

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val logId: Long? = null
) {
    constructor(title: String, contents: String, writer: String, documentBytes: Long, generateTime: LocalDateTime)
            : this(title, contents, writer.toString(), documentBytes, generateTime, null)

    override fun equals(other: Any?): Boolean =
        other is Log && this.logId == other.logId;

    override fun hashCode(): Int = logId?.hashCode() ?: 0
}
